### DISCUSSION TITLE
Sort a Enum value based on description

### DISCUSSION THREAD

**Discussion by EdwinThomas11** [serenity-is/Serenity#7035] [Help] [answered]
public enum UploadStatus
    {
        [Description("Init")]
        Init = 0,
        [Description("ValidationSuccess")]
        ValidationSuccess = 1,
        [Description("ValidationFail")]
        ValidationFail = 2,
        [Description("Deleted")]
        Deleted = 5
    }
    
These are my enumerations; they are stored as integers in the database. I want to perform sorting based on the description within the "process" method. Is it feasible to handle the sorting directly in the "process" method?

@VictorTomaili @volkanceylan please suggest.

**Comment by VictorTomaili** [comment]
To order items you can create an UploadStatusEnumEditor, you need to inherit from EnumEditor and sort the items in the `updateItems()` function. You can refer to the enum editor itself for more details.

**Reply by VictorTomaili** [reply]
You can also create a select2 editor with predefined values that correspond to the enum values.

**Reply by EdwinThomas11** [reply]
If I opt for the UploadStatusEnumEditor, it will only function on sorting that specific column. Is there a way to apply it to all the enums used?

**Reply by VictorTomaili** [reply, marked as answer]
To override the updateItems() function of the EnumEditor, you need to use prototype overriding in TypeScript. You can override this function in the InitScript.ts file.

**Reply by EdwinThomas11** [reply]
Thanks for the quick response @VictorTomaili. Much appreciated.
